pyiron_base/interfaces/has_hdf.py

```diff
@@ -0,0 +1,63 @@
+from abc import ABC, abstractmethod
+
+class WithHDF:
+    __slots__ = ("_hdf", "_group_name")
+
+    def __init__(self, hdf, group_name=None):
+        self._hdf = hdf
+        self._group_name = group_name
+
+    def __enter__(self):
+        if self._group_name is not None:
+            self._hdf = self._hdf.open(self._group_name)
+
+        return self._hdf
+
+    def __exit__(self, *args):
+        if self._group_name is not None:
+            self._hdf.close()
+
+class HasHDF(ABC):
+
+    __version__ = "0.1.0"
+    __hdf_version__ = "0.1.0"
+
+    @abstractmethod
+    def _from_hdf(self, hdf, version=None):
+        pass
+
+    @abstractmethod
+    def _to_hdf(self, hdf):
+        pass
+
+    @abstractmethod
+    def _get_hdf_group_name(self):
```

Comment:
```suggestion
    @property
    @abstractmethod
    def _hdf_group_name(self):
```

At least I think that's the right order for the decorators. 

Am I missing a side effect?

Reply:
It should work, I didn't do it before because it's a private method anyway, but I can change it now.